wavesexchange_topic/src/lib.rs

```diff
@@ -303,10 +321,88 @@ mod parse_and_format {
                             return Err(TopicParseError::InvalidLeasingBalanceTopic);
                         }
                     }
+                    TopicKind::Pairs => {
+                        Topic::extract_exchange_pairs_from_query(&url)?;
+                    }
                 }
 
                 Ok(())
             }
+
+            fn extract_pairs_from_get_params(
+                pairs_in_get_params: Option<Vec<Cow<str>>>,
+            ) -> Result<ExchangePairs, TopicParseError> {
+                // topic://pairs?pairs[]=amount_asset_id/price_asset_id&pairs[]=amount_asset_id1/price_asset_id1
+
+                let mut pairs = vec![];
+
+                match pairs_in_get_params {
+                    None => return Err(TopicParseError::InvalidExchangePairs),
+                    Some(q_pair) => {
+                        for p in q_pair {
+                            let pair = p.split("/").collect::<Vec<&str>>();
```

Comment:
Unnecessary `collect`, since the collected vector is immediately converted to iterator again. Not critical, just suboptimal.

Reply:
done